src/java/org/apache/cassandra/index/sai/disk/v1/sortedterms/SortedTermsWriter.java

```diff
@@ -179,36 +173,29 @@ public void close() throws IOException
     {
         try (IndexOutput output = metadataWriter.builder(componentName))
         {
-            long trieFilePointer = this.trieWriter.complete();
-            SAICodecUtils.writeFooter(trieOutput);
             SAICodecUtils.writeFooter(termsOutput);
-            SortedTermsMeta.write(output, trieFilePointer, pointId, maxLength);
+            SortedTermsMeta.write(output, pointId, maxTermLength);
             // Don't close the offsets writer quietly because of the work it does
             // during its close. We need to propagate the error.
             offsetsWriter.close();
         }
         finally
         {
-            FileUtils.closeQuietly(Arrays.asList(trieWriter, trieOutput, termsOutput));
+            FileUtils.closeQuietly(termsOutput);
```

Comment:
Should we close normally rather than quietly here? i.e. We need the `SequentialWriter` to close properly?